web/packages/teleport/src/Integrations/EditAwsOidcIntegrationDialog.test.tsx

```diff
@@ -0,0 +1,200 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { render, screen, fireEvent } from 'design/utils/testing';
+
+import {
+  Integration,
+  IntegrationKind,
+  IntegrationStatusCode,
+} from 'teleport/services/integrations';
+
+import { EditAwsOidcIntegrationDialog } from './EditAwsOidcIntegrationDialog';
+
+test('edit without s3 fields', async () => {
+  render(
+    <EditAwsOidcIntegrationDialog
+      close={() => null}
+      edit={() => null}
+      integration={{
+        resourceType: 'integration',
+        kind: IntegrationKind.AwsOidc,
+        name: 'some-integration-name',
+        spec: {
+          roleArn: 'arn:aws:iam::123456789012:role/johndoe',
+          s3Bucket: '',
+          s3Prefix: '',
+        },
+        statusCode: IntegrationStatusCode.Running,
+      }}
+    />
+  );
+
+  const checkbox = /I have ran/i;
+  const script = /bash -c/i;
+
+  // Initial state.
+  expect(screen.getByText(/required/i)).toBeInTheDocument();
+  expect(screen.queryByText(script)).not.toBeInTheDocument();
+  expect(screen.queryByText(checkbox)).not.toBeInTheDocument();
```

Comment:
Is there a better way to match these elements vs text? Even resorting to `data-testid` will ensure these tests don't break if any copy changes